### PR TITLE
AMDGPU: Support llvm.exp10

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
ping

File: llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp [hunk 1]
```diff
@@ -2827,12 +2830,59 @@ SDValue AMDGPUTargetLowering::lowerFEXPUnsafe(SDValue X, const SDLoc &SL,
...
+    SDValue K1 = DAG.getConstantFP(0x1.4f0978p-11f, SL, VT);
+
+
+    SDValue Mul0 = DAG.getNode(ISD::FMUL, SL, VT, X, K0, Flags);
+    SDValue Exp2_0 = DAG.getNode(Exp2Op, SL, VT, Mul0, Flags);
+    SDValue Mul1 = DAG.getNode(ISD::FMUL, SL, VT, X, K1, Flags);
+    SDValue Exp2_1 = DAG.getNode(Exp2Op, SL, VT, Mul1, Flags);
+    return DAG.getNode(ISD::FMUL, SL, VT, Exp2_0, Exp2_1);
+  }
+
```
Comment (Pierre-vh) [inline llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:2851]: Can you add a comment like the one in the `VT != MVT::f32` case to show what the math is?

File: llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp [hunk 2]
```diff
@@ -2827,12 +2830,59 @@ SDValue AMDGPUTargetLowering::lowerFEXPUnsafe(SDValue X, const SDLoc &SL,
                      Flags);
 }
 
+SDValue AMDGPUTargetLowering::lowerFEXP10Unsafe(SDValue X, const SDLoc &SL,
```
Comment (Pierre-vh) [inline llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:2833]: What is "unsafe" lowering and how is it different? Is it faster but less accurate?
Can you add a comment to explain that?

**Comment by arsenm** [comment]
ping

**Comment by arsenm** [comment]
ping

**Comment by arsenm** [comment]
ping

**Review by Sisyph — COMMENTED** [review]
I cannot verify whether that approximation is reasonable, but aside from my one comment it looks good.

**Review by Pierre-vh — APPROVED** [review]
LGTM, do we need to add the approximate lowering to GlobalISel as well at some point?
+ run clang-format as well, formatter checks failed

**Comment by arsenm** [comment]
> LGTM, do we need to add the approximate lowering to GlobalISel as well at some point?
> 
> * run clang-format as well, formatter checks failed

I'm ignoring it since it mangles action builder calls badly

**Pull request merged**